### PR TITLE
Get reviewers dynamically for `localise-create-task`

### PR DESCRIPTION
**Pull request by igorDolzh**
https://linear.app/pleo/issue/HMS-304/update-localise-create-task-to-get-reviewers-dynamicly

For `localise-create-task` action, there is a param, where we have to add/update the list of reviewers per language. And if this list should be updated, I have to go at least for 3 projects (in the future it will be more): `product-web`, `pleo-mobile`, `mail-templates` and update manually this list. Also waiting for all checks to be green hoping no flacky tests appear. In other words, it takes time, it feels as redundant time and it is not scalable.

That's why I introduce the solution where the list of reviewers could be updated from the Localise with the help of user groups.

Here is the short process how to deal with such updates:
https://www.notion.so/pleo/Create-Update-list-of-the-translators-61a8a96bf80e4cfb9195f7c7f05a57e4

Later on I will update `product-web`, `pleo-mobile` and `mail-templates`

### CODE REVIEW

**Review by dariye — APPROVED** [review]

File: lokalise-create-task/lib/main.js [hunk 1]
```diff
@@ -41,24 +41,42 @@ function getKeys(lokalise) {
...
+                const userGroupsWithLanguage = userGroups.filter((userGroupDetails) => {
+                    const languageListInPermissions = userGroupDetails.permissions.languages.map(({ lang_iso: langISO }) => langISO);
+                    return languageListInPermissions.includes(language);
+                });
+                const members = userGroupsWithLanguage.reduce((acc, userGroup) => {
+                    return [...acc, ...userGroup.members];
+                }, []);
+                if (members &&
+                    members.length > 0 &&
+                    supportedLanguagesParsed.includes(language)) {
```
Comment (braedongough) [inline lokalise-create-task/lib/main.js:73]: Maybe we could store this in a more descriptive variable to improve readability like `membersSupportLanguagesParsed`

**Review by braedongough — APPROVED** [review]

File: lokalise-create-task/src/main.ts [hunk 2]
```diff
@@ -26,28 +26,62 @@ async function getKeys(lokalise: LokaliseApi) {
...
-      if (assigneeList && assigneeList.length > 0) {
-        assigneeIdObject[key] = assigneeList.map(({ user_id }) => user_id);
+      const members: number[] = userGroupsWithLanguage.reduce(
+        (acc: number[], userGroup: { members: number[] }) => {
+          return [...acc, ...userGroup.members];
+        },
+        []
+      );
+      if (
+        members &&
```
Comment (pekala) [inline lokalise-create-task/src/main.ts:75]: I guess this is unnecessary, `members` will always be at least an empty array?

File: lokalise-create-task/src/main.ts [hunk 3]
```diff
@@ -26,28 +26,62 @@ async function getKeys(lokalise: LokaliseApi) {
...
-    const teamUsers = await lokalise.teamUsers.list({ team_id: team.team_id });
-    const assigneeEmailListParsed = JSON.parse(assigneeEmailList);
+    const userGroups = await lokalise.userGroups.list({
+      team_id: team.team_id,
+    });
 
-    const assigneeIdObject = {} as any;
-    Object.keys(assigneeEmailListParsed).map(function (key: string) {
-      const assigneeEmailListPerLanguage = assigneeEmailListParsed[key];
+    const assigneeIdObject = {} as { [language: string]: number[] };
```
Comment (pekala) [inline lokalise-create-task/src/main.ts:52]: A bit more idomatic
```suggestion
    const assigneeIdObject:  { [language: string]: number[] }; = {}
```

File: lokalise-create-task/src/main.ts [hunk 4]
```diff
@@ -26,28 +26,62 @@ async function getKeys(lokalise: LokaliseApi) {
...
+/**
+ * Returns an object languages as key and value is the list of ids of the members for certain language
+ */
+async function getAssigneeIdObject(
+  lokalise: LokaliseApi
+): Promise<{ [language: string]: number[] }> {
   const teams = await lokalise.teams.list();
-  const team = teams.find(({ name }) => name === teamName);
+  const supportedTaskLanguagesParsed = supportedLanguages
+    ? JSON.parse(supportedLanguages)
```
Comment (pekala) [inline lokalise-create-task/src/main.ts:37]: If you want to be nice, you could wrap the parse in try catch and produce a meaningful error when the JSON passed is malformed

**Review by pekala — APPROVED** [review]
Makes sense!

File: lokalise-create-task/src/main.ts (hunk 4 above)
Comment (igorDolzh) [reply lokalise-create-task/src/main.ts:37]: Yeah, good suggestion, I will properly do that for all of "localise" actions.
https://linear.app/pleo/issue/HMS-340/wrap-jsonparse-into-meaningful-trycatch-for-localise-actions

**Pull request merged**